src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -86,7 +86,9 @@
     ["extract" columns crit]
 
     [["extract" columns subquery]]
-    ["extract" columns ["and" subquery crit]]
+    (if (= "group_by" (first subquery))
```

Comment:
I had that but changed it to this, because it just leads to duplicating the clause and a guard. Thought this looked cleaner.